auxx/src/Command/Tx.hs

```diff
@@ -92,28 +95,58 @@ sendToAllGenesis diffusion (SendToAllGenesisParams txsPerThread conc delay_ tpsS
+        -- and a queue with necessary data for the creation of txs whose inputs
+        -- don't belong in genesis block
+        txQueue            <- atomically $ newTQueue
+        txPreparationQueue <- atomically $ newTQueue
         logInfo $ sformat ("Found "%shown%" keys in the genesis block.") (length keysToSend)
         startAtTxt <- liftIO $ lookupEnv "AUXX_START_AT"
         let startAt = fromMaybe 0 . readMaybe . fromMaybe "" $ startAtTxt :: Int
-        -- construct transaction output
-        outAddr <- makePubKeyAddressAuxx (toPublic (fromMaybe (error "sendToAllGenesis: no keys") $ head keysToSend))
-        let txOut1 = TxOut {
-                txOutAddress = outAddr,
-                txOutValue = mkCoin 1
-                }
-            txOuts = TxOutAux txOut1 :| []
         -- construct a transaction, and add it to the queue
         let addTx secretKey = do
-                utxo <- getOwnUtxoForPk $ safeToPublic (fakeSigner secretKey)
-                etx <- createTx mempty utxo (fakeSigner secretKey) txOuts (toPublic secretKey)
+                let signer = fakeSigner secretKey
+                    publicKey = toPublic secretKey
+                -- construct transaction output
+                outAddr <- makePubKeyAddressAuxx publicKey
+                let txOut1 = TxOut {
+                    txOutAddress = outAddr,
+                    txOutValue = mkCoin 1
+                    }
+                    txOuts = TxOutAux txOut1 :| []
+                utxo <- getOwnUtxoForPk $ safeToPublic signer
+                etx <- createTx mempty utxo signer txOuts publicKey
                 case etx of
                     Left err -> logError (sformat ("Error: "%build%" while trying to contruct tx") err)
-                    Right (tx, _) -> atomically $ writeTQueue txQueue (tx, txOuts)
-        let nTrans = conc * txsPerThread
-            allTrans = take nTrans (drop startAt keysToSend)
-            (firstBatch, secondBatch) = splitAt ((2 * nTrans) `div` 3) allTrans
+                    Right (tx, _) -> do
+                        atomically $ writeTQueue txQueue tx
+                        atomically $ writeTQueue txPreparationQueue (tx, txOut1, secretKey)
+        let genesisTxs  = conc * genesisTxsPerThread
+            nTxs        = conc * txsPerThread
+            genesisKeys = take genesisTxs (drop startAt keysToSend)
+            -- construct transactions whose inputs does not belong in genesis
+            -- block. Send as many coins as were received in the previous round
+            -- back to yourself.
+            prepareTxs :: Int -> m ()
+            prepareTxs n
+                | n <= 0 = return ()
+                | otherwise = (atomically $ tryReadTQueue txPreparationQueue) >>= \case
+                    Just (tx, txOut1', senderKey) -> do
+                        let txInp = TxInUtxo (hash (taTx tx)) 0
+                            utxo' = M.fromList [(txInp, TxOutAux txOut1')]
```

Comment:
So, what you're doing here is _expecting_ that the output from the previously sent transaction is included in the UTxO by the time this transaction will actually be sent, right?
Might be worth to add a comment explaining that.